homeassistant/components/twitch/__init__.py

```diff
@@ -17,11 +17,13 @@
     async_get_config_entry_implementation,
 )
 
-from .const import DOMAIN, OAUTH_SCOPES, PLATFORMS
+from .const import OAUTH_SCOPES, PLATFORMS
 from .coordinator import TwitchCoordinator
 
+type TwitchConfigEntry = ConfigEntry[TwitchCoordinator]
```

Comment:
I suggest to also pass the config entry explicitly in the coordinator (and probably move the definition to `coordinator.py` to avoid circular imports)